pipeline/state_duration.go

```diff
@@ -0,0 +1,58 @@
+package pipeline
+
+import (
+	"time"
+
+	"github.com/influxdata/kapacitor/tick/ast"
+)
+
+// Compute the duration of a given state.
+// The state is defined via a lambda expression. For each consecutive point for
+// which the expression evaluates as true, the state duration will be
+// incremented by the duration between points. When a point evaluates as false,
+// the state duration is reset to 0.
+//
+// The state duration will be added as an additional field to each point. If the
+// expression evaluates as false, the field will not be added. If the expression
```

Comment:
OK, so why is it important to distinguish between the first point that evaluates to true and other points that evaluate to false? My understanding was that the user cares about the duration of the event and so the user wouldn't trigger anything on the first point anyways. 

Said a different way under what circumstances would the user want to trigger an alert when the stateDuration is 0? Seems like the alert condition would always be "state_duration" > 0 . But even if not you could always reevaluate the same state later if it was important. 

For example:

```
stream
  |from().measurement('ntpq')|default().tag('state_prefix',' ')
  |groupBy('host')
  |stateDuration(lambda: "state_prefix" != '*').unit(1m)
  |default().field('state_duration',-1)
  |alert()
    // Trigger info on first occurrence.
    .info(lambda: "state_duration" == 0 AND "state_prefix" != '*') 
    // Warn after 5 minutes.
    .warn(lambda: "state_duration" > 5)
    // Critical after 1 hour.
    .crit(lambda: "state_duration" > 60)
```

From my perspective it seems that the node would be easier to reason about if it always set the field with either a 0 or the current duration, but I want to understand your perspective.